platform/android/export/export.cpp

```diff
@@ -2536,63 +2545,90 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		EditorProgress ep("export", "Exporting for Android", 105, true);
 
 		bool use_custom_build = bool(p_preset->get("custom_template/use_custom_build"));
+		int export_format = int(p_preset->get("custom_template/export_format"));
 		bool p_give_internet = p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG);
+		bool _signed = p_preset->get("package/signed");
+		bool apk_expansion = p_preset->get("apk_expansion/enable");
+		Vector<String> enabled_abis = get_enabled_abis(p_preset);
 
 		Ref<Image> main_image;
 		Ref<Image> foreground;
 		Ref<Image> background;
 
 		load_icon_refs(p_preset, main_image, foreground, background);
 
-		if (use_custom_build) {
-			//re-generate build.gradle and AndroidManifest.xml
-			{ //test that installed build version is alright
-				FileAccessRef f = FileAccess::open("res://android/.build_version", FileAccess::READ);
-				if (!f) {
-					EditorNode::get_singleton()->show_warning(TTR("Trying to build from a custom built template, but no version info for it exists. Please reinstall from the 'Project' menu."));
-					return ERR_UNCONFIGURED;
-				}
-				String version = f->get_line().strip_edges();
-				if (version != VERSION_FULL_CONFIG) {
-					EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
-					return ERR_UNCONFIGURED;
-				}
+		Vector<uint8_t> command_line_flags;
+		// Write command line flags into the command_line_flags variable.
+		Error err = get_command_line_flags(p_preset, p_path, p_flags, command_line_flags);
+		if (err != OK) {
+			EditorNode::add_io_error("Could not generate command line file.");
+			return ERR_SKIP;
+		}
+
+		if (export_format == 1 && !p_path.ends_with(".aab")) {
+			EditorNode::get_singleton()->show_warning(TTR("Invalid filename! Android App Bundle requires the *.aab extension."));
+			return ERR_UNCONFIGURED;
+		}
+		if (export_format == 0 && !p_path.ends_with(".apk")) {
+			EditorNode::get_singleton()->show_warning(
+					TTR("Invalid filename! Android APK requires the *.apk extension."));
+			return ERR_UNCONFIGURED;
+		}
+		if (export_format > 1 || export_format < 0) {
+			EditorNode::add_io_error("Unsupported export format!\n");
+			return ERR_UNCONFIGURED; //TODO: is this the right error?
+		}
+
+		if (use_custom_build && !apk_expansion) {
```

Comment:
For now, I've made it so that apk expansion is only allowed when use custom build is disabled. I think this might need to be changed.